app/controllers/concerns/ab_testing_concern.rb

```diff
@@ -0,0 +1,18 @@
+# frozen_string_literal: true
+
+module AbTestingConcern
+  # @param [Symbol] test Name of the test, which should correspond to an A/B test defined in
+  # #                    config/initializer/ab_tests.rb.
+  # @return [Symbol,nil] Bucket to use for the given test, or nil if the test is not active.
+  def ab_test_bucket(test_name)
+    test = AbTests.all[test_name]
+    raise "Unknown A/B test: #{test_name}" unless test
+
+    test.bucket(
```

Comment:
`Hash#fetch` will throw a `KeyError` if the key is not found, could remove some custom error handling

```suggestion
    AbTests.all.fetch(test_name).bucket(
```